docs/src/modules/concepts/pages/concepts.adoc

```diff
@@ -17,16 +17,60 @@ Production-ready:: No code changes required to move from development to producti
+image:concepts:agentic-ai-system-anatomy.png[Akka Agentic Platform]
+
+[cols="1,3", options="header"]
+|===
+|Aspect |AI Role and Responsibility
+
+|Agents
+|Components that integrate with AI to perceive their environment, make decisions and take actions toward a specific goal
+
+You implement agents in Akka with the xref:java:agents.adoc[Agents] component.
+
+|Tools
+|Any public method made available to Model Context Protocol (MCP) clients.
+
+You invoke tools in Akka through _embedded agent function calls_ or by _invoking a remote MCP tool_. You can implement MCP servers with the xref:java:mcp-endpoints.adoc[MCP Endpoints] component.
+
+|Endpoints
+|Externally accessible entry points through which agents are launched and controlled.
+
+You implement Endpoints in Akka using either xref:java:http-endpoints.adoc[HTTP], xref:java:grpc-endpoints.adoc[gRPC] or xref:java:mcp-endpoints.adoc[MCP] protocols.
+
+|Goals
+|Clear objectives or outcomes that agents continuously work toward by making decisions and taking actions on their own.
+
+You implement goals in Akka by implementing a multi-agent system _with a planner agent_ using a xref:java:workflows.adoc[Workflow] component to orchestrate the cross-agent interactions.
+
+|Guardians
+|Components that monitor, protect and evaluate the system against its goals and constraints. 
+
+You implement guardians in Akka by...
+ifdef::todo[TODO: get some thoughts from Patrik on this]
+|Adaptation
+|Continuous, real-time streams from users or the environment which can alter the context, memory or semantic knowledge used by an agentic system.
+
+You implement adaptation in Akka by processing a stream of data from external sensors, either with the xref:java:consuming-producing.adoc[Consumer] component or through streaming HTTP or gRPC interfaces. xref:java:consuming-producing.adoc[Consumers] can modify an agent’s goals, memory, or guardians to affect the behavior of the system.
+
+|Orchestration
+|The ability to execute, persist and recover long-running tasks made possible through _durable execution_.
+
+You implement orchestration in Akka with the xref:java:workflows.adoc[Workflow] component.
+
+|Memory
+|Data that enables agents to reason over time, track context, make correct decisions and learn from experience.
+
+You inherit agentic and episodic (short-term) durable memory automatically when you implement a stateful xref:java:agents.adoc[Agent] component. You can get long-term, multi-agent memory by implementing xref:java:event-sourced-entities.adoc[Event Sourced Entity] or xref:java:key-value-entities.adoc[Key Value Entity] components.
+
+|Registry
+|A built-in directory that stores information about all agents so they can be discovered and called upon in multi-agent systems.
+
+You implement a registry in Akka by... 
+ifdef::todo[TODO: We need to talk to Patrik about this...]
```

Comment:
@patriknw Tyler asked for your input here

Reply:
You don't implement the agent registry. It's provided by the SDK, but you would give the agent components meaningful names and descriptions with an annotation. It's mentioned here https://doc.akka.io/snapshots/akka-documentation/java/agents.html#_creating_dynamic_plans